vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go

```diff
@@ -67,3 +112,59 @@ func ObserveUsageRecommendationRelativeDiff(usage, recommendation float64, isOOM
 func ObserveMissingRecommendation(isOOM bool, resource corev1.ResourceName) {
 	usageMissingRecommendationCounter.WithLabelValues(string(resource), strconv.FormatBool(isOOM)).Inc()
 }
+
+// ObserveUsageRecommendationDiff records absolute diff between usage and
+// recommendation.
+func ObserveUsageRecommendationDiff(usage, recommendation float64, isOOM bool, resource corev1.ResourceName) {
+	switch resource {
+	case corev1.ResourceCPU:
+		cpuUsageRecommendationDiff.Observe((usage - recommendation))
+	case corev1.ResourceMemory:
+		memoryUsageRecommendationDiff.WithLabelValues(strconv.FormatBool(isOOM)).Observe(usage - recommendation)
+	default:
+		klog.Warningf("Unknown resource: %v", resource)
+	}
+}
+
+// ObserveRecommendation records the value of recommendation.
+func ObserveRecommendation(recommendation float64, isOOM bool, resource corev1.ResourceName) {
```

Comment:
Have you considered not exporting this function (and 3 previous ones)?
I understand the two unchanged functions were used directly in the previous PR but now we have the ObserveQualityMetrics* as public API.

Reply:
Done.